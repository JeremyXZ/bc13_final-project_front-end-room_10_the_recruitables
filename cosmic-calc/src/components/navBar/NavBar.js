import React from "react";
import { Nav, NavLink, NavMenu } from "./navBarElements";

const NavBar = () => {
  return (
    <>
      <Nav>
        <NavMenu>
          <NavLink to="/login" activeStyle>
            login
          </NavLink>
          <NavLink to="/profile" activeStyle>
            profile
          </NavLink>
          <NavLink to="/progress" activeStyle>
            main
          </NavLink>
          <NavLink to="/game" activeStyle>
            game
          </NavLink>
        </NavMenu>
      </Nav>
    </>
  );
};

export default NavBar;
