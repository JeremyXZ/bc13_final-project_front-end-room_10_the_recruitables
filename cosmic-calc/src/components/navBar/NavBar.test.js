import React from "react";
import "@testing-library/jest-dom";
import {screen, render } from "@testing-library/react";
import {BrowserRouter, Router} from 'react-router-dom';
import userEvent from "@testing-library/user-event";

import NavBar from "./NavBar";

describe("<NavBar/>", () => {
    it("successfully renders the component, matches snapshot", () => {
        const { asFragment } = render(
            <BrowserRouter>
                <NavBar />
            </BrowserRouter>
        
        );
        expect(asFragment()).toMatchSnapshot();
      });

    it('renders 4 links on the screen', () => {
        render(
            <BrowserRouter>
                <NavBar />
            </BrowserRouter>
        
        );
       
        const links = screen.getAllByRole('link')
        expect(links.length).toBe(4)
    })

    it('renders all 3 link texts on the screen', () => {
        const {getAllByText} = render(
            <BrowserRouter>
                <NavBar />
            </BrowserRouter>
        
        );

   
    expect(getAllByText('Profile')[0]).toBeInTheDocument();
    expect(getAllByText('Progress')[0]).toBeInTheDocument();
    expect(getAllByText('Game')[0]).toBeInTheDocument();
    
    })

})