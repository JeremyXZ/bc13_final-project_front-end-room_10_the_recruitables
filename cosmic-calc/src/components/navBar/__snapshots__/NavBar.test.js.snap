// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar/> successfully renders the component, matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="nav-bar"
  >
    <nav
      class="sc-bcXHqe llTDYN"
    >
      <img
        alt="logo"
        class="logo"
        src="logo.png"
      />
      <div
        class="sc-dkrFOg fLDoxR"
      >
        <a
          class="sc-gswNZR kjkvCA"
          href="/profile"
        >
          Profile
        </a>
        <a
          class="sc-gswNZR kjkvCA"
          href="/progress"
        >
          Progress
        </a>
        <a
          class="sc-gswNZR kjkvCA"
          href="/game"
        >
          Game
        </a>
        <a
          aria-current="page"
          class="sc-gswNZR kjkvCA active"
          href="/"
        >
          <button
            class="navButtonLogout"
          />
        </a>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;
